service/controller/resource/app/create.go

```diff
@@ -0,0 +1,41 @@
+package app
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/giantswarm/cleanup-operator/service/controller/key"
+	"github.com/giantswarm/microerror"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	app, err := key.ToApp(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	logger := r.logger.With("namespace", app.Namespace, "app", app.Name)
```

Comment:
There is no need for that. All that information is available in `object` key which is set by the operatorkit.

Reply:
I think this is the relevant part https://github.com/giantswarm/operatorkit/blob/83e206bee8c92ad61539e340b808f34b9ed98bfd/controller/controller.go#L440-L442